src/sql/base/browser/ui/button/button.ts

```diff
@@ -13,11 +13,10 @@ export interface IButtonStyles extends vsIButtonStyles {
 }
 
 export class Button extends vsButton {
-	private buttonFocusOutline: Color;
+	private buttonFocusOutline?: Color;
```

Comment:
VS Code seems to prefer using types like `Color | null`. Does it make a difference?

Reply:
They do it differently in different areas. HTMLElement style properties are defined as `type | null` so you can't assign them to be `undefined` (`undefined != null`). But most places we use undefined, and the `?` specifier to translated as `undefined` not `null` so in places where you want to directly assign a variable to a html property it can be cleaner to use `Color | null` otherwise it's cleaner to use `?: Color`.

Reply:
Specifically for `color` since you never set a html property to it directly, its probably always correct to use `?: Color`. But I think with vscode it mainly comes down to who wrote it.
https://github.com/Microsoft/azuredatastudio/blob/master/src/vs/editor/contrib/gotoError/gotoErrorWidget.ts#L167
https://github.com/Microsoft/azuredatastudio/blob/master/src/vs/base/browser/ui/menu/menu.ts#L36
or even
`?: Color | null`
https://github.com/Microsoft/azuredatastudio/blob/master/src/vs/editor/contrib/zoneWidget/zoneWidget.ts#L34

Reply:
From a javascript point of view `undefined` means the value was never set where as `null` is meaning you set the value to not exist. But it's really just semantics.
https://www.ajaymatharu.com/javascript-difference-between-undefined-and-null/